File: Test_gan_DoiTuong_dt2_cho_DoiTuong_dt1_code_Cpp/Person.cpp
#include <iostream>
using namespace std;

class Person{
    private:
        int id;
        string ten;

    public:
        Person(){
            id = 0;
            ten = "NO NAME";
        }

    public:
        Person(int data1, string data2){
            id = data1;
            ten = data2;
        }

    public:
        ~Person(){}

    // hàm gán đối tượng
    // gọi là hàm copy đối tượng cũng được
    public:
        Person(Person &dt){
            id = dt.get_id();
            ten = dt.get_ten();
        }

    // hiển thị
    public:
        void HienThi(){
            cout << id << "\t" + ten + "\n";
        }

    // get_id và get_ten
    public:
        int get_id(){
            return id;
        }

        string get_ten(){
            return ten;
        }

    // set_id và set_ten
    public:
        void set_id(int data){
            id = data;
        }

        void set_ten(string data){
            ten = data;
        }
};

File: Test_gan_DoiTuong_dt2_cho_DoiTuong_dt1_code_Java/Person.java
public class Person {
    private int id;
    private String ten;

    // hàm khởi tạo không tham số
    public Person(){
        id = 0;
        ten = "NO NAME";
    }

    // hàm khởi tạo có tham số
    public Person(int data1, String data2){
        id = data1;
        ten = data2;
    }

    public void HienThi(){
        System.out.print(id + "\t" + ten + "\n");
    }

    // get_id và get_ten
    public int get_id(){
        return id;
    }
    public String get_ten(){
        return ten;
    }

    // set_id và set_ten
    public void set_id(int data){
        id = data;
    }
    public void set_ten(String data){
        ten = data;
    }

    // hàm copy trong java
    // tức là lấy dt2 gán cho dt1
    public Person(Person dt){
        id = dt.get_id();
        ten = dt.get_ten();
    }
}
